test/OpenSpaceToolkit/Astrodynamics/Estimator/ODLeastSquaresSolver.test.cpp

```diff
@@ -0,0 +1,265 @@
+        }
+
+        positionSigmas_ << 1.0e1, 1.0e1, 1.0e1;
+        velocitySigmas_ << 1.0e-2, 1.0e-2, 1.0e-2;
+
+        referenceStateSigmas_ = {
+            {*CartesianPosition::Default(), positionSigmas_},
+            {*CartesianVelocity::Default(), velocitySigmas_},
+        };
+        initialStateSigmas_ = {
+            {*CartesianPosition::Default(), positionSigmas_},
+            {*CartesianVelocity::Default(), velocitySigmas_},
+        };
+    }
+
+    const Earth earth_ = Earth::EGM96(10, 10);
+
+    Environment environment_ = Environment(std::make_shared<Earth>(earth_), {});
+    NumericalSolver numericalSolver_ = NumericalSolver::Default();
+    LeastSquaresSolver leastSquaresSolver_ = LeastSquaresSolver::Default();
+    const ODLeastSquaresSolver odSolver_ = {environment_, numericalSolver_, leastSquaresSolver_};
+
+    Array<State> referenceStates_ = Array<State>::Empty();
+
+    VectorXd positionSigmas_ {3};
+    VectorXd velocitySigmas_ {3};
+
+    std::unordered_map<CoordinateSubset, VectorXd> referenceStateSigmas_ = {};
+    std::unordered_map<CoordinateSubset, VectorXd> initialStateSigmas_ = {};
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_ODLeastSquaresSolver, Constructor)
+{
+    {
+        EXPECT_NO_THROW(ODLeastSquaresSolver(environment_, numericalSolver_));
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_ODLeastSquaresSolver, Accessors)
+{
+    {
+        const ODLeastSquaresSolver solver(environment_, numericalSolver_);
+
+        EXPECT_EQ(solver.accessEnvironment().accessCentralCelestialObject()->getName(), "Earth");
+        EXPECT_TRUE(solver.accessPropagator().isDefined());
+        EXPECT_NO_THROW(solver.accessSolver());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_ODLeastSquaresSolver, EstimateState)
+{
+    {
+        // Test basic state estimator
+        const auto analysis = odSolver_.estimateState(referenceStates_[0], referenceStates_);
+
+        EXPECT_EQ(analysis.getSolverAnalysis().getTerminationCriteria(), "RMS Update Threshold");
+        EXPECT_LT(analysis.getSolverAnalysis().getRmsError(), 50.0);
+    }
+
+    {
+        // Test with estimator coordinate subsets
+        Array<Shared<const CoordinateSubset>> estimationSubsets = {CartesianPosition::Default()};
+
+        const auto analysis = odSolver_.estimateState(referenceStates_[0], referenceStates_, estimationSubsets);
+
+        EXPECT_EQ(analysis.getSolverAnalysis().getTerminationCriteria(), "RMS Update Threshold");
+        EXPECT_LT(analysis.getSolverAnalysis().getRmsError(), 50.0);
+    }
+
+    {
+        // Test with sigmas
+        const auto analysis = odSolver_.estimateState(
+            referenceStates_[0], referenceStates_, {}, initialStateSigmas_, referenceStateSigmas_
+        );
+
+        EXPECT_EQ(analysis.getSolverAnalysis().getTerminationCriteria(), "RMS Update Threshold");
+        EXPECT_LT(analysis.getSolverAnalysis().getRmsError(), 50.0);
+    }
+}
```

Comment:
Could we add something like this:

```suggestion
    {
        // Test basic state estimator with miscellaneous frames
        const State initialGuessStateInTEME = referenceStates_[0].inFrame(Frame::TEME());
        const Array<State> referenceStatesInITRF = referenceStates_.map<State>(
            {
                return aState.inFrame(Frame::ITRF());
            }
        );
        
        const auto analysis = odSolver_.estimateState(referenceStates_[0], referenceStates_);
        const auto analysisInMiscellaneousFrames = odSolver_.estimateState(initialGuessStateInTEME, referenceStatesInITRF);

        EXPECT_EQ(analysisInMiscellaneousFrames.getSolverAnalysis().getTerminationCriteria(), analysis.getSolverAnalysis().getTerminationCriteria());
        EXPECT_EQ(analysisInMiscellaneousFrames.getSolverAnalysis().getRmsError(), analysis.getSolverAnalysis().getRmsError());  // Or assert that is close somehow
    }
}
```